classes/userGroup/Repository.php

```diff
@@ -0,0 +1,469 @@
+
+    /** @var Request */
+    protected $request;
+
+    /** @var PKPSchemaService */
+    protected $schemaService;
+
+    public function __construct(DAO $dao, Request $request, PKPSchemaService $schemaService)
+    {
+        $this->dao = $dao;
+        $this->request = $request;
+        $this->schemaService = $schemaService;
+    }
+
+    /** @copydoc DAO::newDataObject() */
+    public function newDataObject(array $params = []): UserGroup
+    {
+        $object = $this->dao->newDataObject();
+        if (!empty($params)) {
+            $object->setAllData($params);
+        }
+        return $object;
+    }
+
+    /** @copydoc DAO::get() */
+    public function get(int $id): ?UserGroup
+    {
+        return $this->dao->get($id);
+    }
+
+    /** @copydoc DAO::getCollector() */
+    public function getCollector(): Collector
+    {
+        return app(Collector::class);
+    }
+
+    /**
+     * Get an instance of the map class for mapping
+     * authors to their schema
+     */
+    public function getSchemaMap(): maps\Schema
+    {
+        return app('maps')->withExtensions($this->schemaMap);
+    }
+
+    /**
+     * Validate properties for an author
+     *
+     * Perform validation checks on data used to add or edit an author.
+     *
+     * @param UserGroup|null $author The author being edited. Pass `null` if creating a new author
+     * @param array $props A key/value array with the new data to validate
+     * @param array $allowedLocales The context's supported submission locales
+     * @param string $primaryLocale The submission's primary locale
+     *
+     * @return array A key/value array with validation errors. Empty if no errors
+     */
+    public function validate($userGroup, $props, $allowedLocales, $primaryLocale)
+    {
+        $schemaService = Services::get('schema');
+
+        $validator = ValidatorFactory::make(
+            $props,
+            $schemaService->getValidationRules(PKPSchemaService::SCHEMA_USER_GROUP, $allowedLocales)
+        );
+
+        // Check required fields
+        ValidatorFactory::required(
+            $validator,
+            $userGroup,
+            $schemaService->getRequiredProps(PKPSchemaService::SCHEMA_USER_GROUP),
+            $schemaService->getMultilingualProps(PKPSchemaService::SCHEMA_USER_GROUP),
+            $allowedLocales,
+            $primaryLocale
+        );
+
+        // Check for input from disallowed locales
+        ValidatorFactory::allowedLocales($validator, $schemaService->getMultilingualProps(PKPSchemaService::SCHEMA_USER_GROUP), $allowedLocales);
+
+        $errors = [];
+        if ($validator->fails()) {
+            $errors = $schemaService->formatValidationErrors($validator->errors());
+        }
+
+        HookRegistry::call('UserGroup::validate', [$errors, $userGroup, $props, $allowedLocales, $primaryLocale]);
+
+        return $errors;
+    }
+
+    /**
+     * @copydoc \PKP\services\entityProperties\EntityWriteInterface::add()
```

Comment:
These `copydoc`s don't add anything; they can be removed.

Reply:
Removed from 
- public function add
- public function edit
- public function delete